staging/src/k8s.io/apiserver/pkg/util/proxy/streamtunnel.go

```diff
@@ -352,11 +357,13 @@ func (u *tunnelingWebsocketUpgraderConn) InitializeWrite(backendResponse *http.R
 		}
 
 		klog.V(4).Infof("websocket connection created: %s", conn.Subprotocol())
+		metrics.IncStreamTunnelRequest(context.Background(), strconv.Itoa(http.StatusOK))
```

Comment:
there is an exit on line 356, should this metric incremented there too?

Reply:
Yes--you are correct. There should be a 500 internal server error metric incremented in this case.

This case represents the situation where the SPDY upgrade was successful, but the subsequent WebSocket upgrade failed.

Reply:
should this be recording StatusOK or StatusSwitchingProtocols?